src/api/Feed.js

```diff
@@ -772,20 +771,31 @@ class Feed extends Base {
                     return;
                 }
 
+                this.addPendingItem(this.file.id, currentUser, pendingTask);
```

Comment:
this line seems suspicious. previously, we did this in `successCallback` after calling the old `.createTask`

Reply:
Ya, you are right. I should move this back inside the successCallBack so that if the createTaskWithDeps API call fails, we won't end up with a leftover pendingItem.